encoding/protojson/well_known_types.go

```diff
@@ -176,7 +176,7 @@ func (d decoder) unmarshalAny(m protoreflect.Message) error {
 	// Use another decoder to parse the unread bytes for @type field. This
 	// avoids advancing a read from current decoder because the current JSON
 	// object may contain the fields of the embedded type.
-	dec := decoder{d.Clone(), UnmarshalOptions{RecursionLimit: d.opts.RecursionLimit}}
+	dec := decoder{d.Clone(), UnmarshalOptions{RecursionLimit: d.opts.RecursionLimit, AllowPartial: d.opts.AllowPartial}}
```

Comment:
Not sure if I should also plumb through DiscardUnknown (or maybe all the options?).  Those are the two options that cloud.google.com/go passes today: e.g. https://github.com/googleapis/google-cloud-go/blob/b23a08492a7724ce6f1f95395b7f71feb9b3de66/ai/generativelanguage/apiv1beta/generative_client.go#L600